src/test/scala/BankingAppDataFrameTest.scala

```diff
@@ -0,0 +1,58 @@
+package com.scalaSparkSkillValidation
+
+import org.apache.spark.sql.SparkSession
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import com.scalaSparkSkillValidation.utils.DataGenerator
+import com.scalaSparkSkillValidation.BankingAppDataFrameImplicits._
+import org.scalatest.BeforeAndAfterAll
+
+class BankingAppDataFrameTest extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
+
+  implicit var spark: SparkSession = _
+
+  /**
+   * Create spark session for test suite
+   * */
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    spark = SparkSession.builder()
+      .master("local[*]")
+      .appName("test")
+      .getOrCreate()
+    setupTestData
+  }
+
+  /**
+   * Tears down the spark session
+   * */
+  override def afterAll(): Unit = {
+    if (spark != null) {
```

Comment:
Yeah, this can be done better in Scala, if U join our team, I'll show U how

Reply:
Awesome, is there any resource you recommend for this?